src/survey_assist_utils/evaluation/coder_alignment.py

```diff
@@ -0,0 +1,550 @@
+            stats.append(
+                {
+                    "threshold": threshold,
+                    "accuracy": self.get_accuracy(threshold),
+                    "coverage": self.get_coverage(threshold),
+                }
+            )
+
+        return pd.DataFrame(stats)
+
+    def plot_threshold_curves(
+        self,
+        thresholds: Optional[list[float]] = None,
+        figsize: tuple[int, int] = (10, 6),
+    ) -> None:
+        """Plot accuracy and coverage curves against confidence threshold.
+
+        Args:
+            thresholds (list[float], optional): list of threshold values to evaluate.
+                If None, default thresholds will be used.
+            figsize (tuple[int, int], optional): Size of the figure in inches (width, height).
+                Defaults to (10, 6).
+        """
+        stats_df = self.get_threshold_stats(thresholds)
+
+        plt.figure(figsize=figsize)
+        plt.plot(
+            stats_df["threshold"],
+            stats_df["coverage"],
+            label="Coverage",
+            color="blue",
+        )
+        plt.plot(
+            stats_df["threshold"],
+            stats_df["accuracy"],
+            label="Accuracy",
+            color="orange",
+        )
+
+        plt.xlabel("Confidence threshold")
+        plt.ylabel("Percentage")
+        plt.grid(True)
+        plt.legend()
+        plt.title("Coverage and Accuracy vs Confidence Threshold")
+        plt.tight_layout()
+        plt.show()
+
+    def get_summary_stats(self) -> dict:
+        """Get summary statistics for the classification results.
+
+        Returns:
+        -------
+            dictionary containing various summary statistics
+        """
+        return {
+            "total_samples": len(self.df),
+            "overall_accuracy": self.get_accuracy(),
+            "accuracy_above_0.50": self.get_accuracy(0.5),
+            "accuracy_above_0.60": self.get_accuracy(0.6),
+            "accuracy_above_0.70": self.get_accuracy(0.7),
+            "accuracy_above_0.80": self.get_accuracy(0.8),
+            "coverage_above_0.50": self.get_coverage(0.5),
+            "coverage_above_0.60": self.get_coverage(0.6),
+            "coverage_above_0.70": self.get_coverage(0.7),
+            "coverage_above_0.80": self.get_coverage(0.8),
+        }
+
+    def plot_confusion_heatmap(
+        self,
+        human_code_col: str,
+        llm_code_col: str,
+        top_n: int = 10,
+        exclude_patterns: Optional[list[str]] = None,
+    ) -> Optional[plt.Axes]:
+        """Generates and displays a confusion matrix heatmap for the top N codes.
+
+        Args:
+            human_code_col (str): The column name for the ground truth codes.
+            llm_code_col (str): The column name for the model's predicted codes.
+            top_n (int): The number of most frequent codes to include in the matrix.
+            exclude_patterns (list[str]): A list of substrings to filter out from the
```

Comment:
Good to catch here, but I'm thinking the values to filter out should be removed already when you calculate get_accuracy. As in, input for confusion matrix should have the same dimension as what was used to calculate accuracy.